server/src/main/java/io/crate/expression/symbol/DynamicReference.java

```diff
@@ -21,29 +21,44 @@
 
 package io.crate.expression.symbol;
 
-import io.crate.metadata.IndexType;
 import io.crate.metadata.SimpleReference;
 import io.crate.metadata.ReferenceIdent;
 import io.crate.metadata.RowGranularity;
 import io.crate.sql.tree.ColumnPolicy;
 import io.crate.types.DataType;
 import io.crate.types.DataTypes;
+
+import org.elasticsearch.Version;
 import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 
 import java.io.IOException;
+import java.util.Objects;
 
 public class DynamicReference extends SimpleReference {
 
-    public DynamicReference(StreamInput in) throws IOException {
-        super(in);
-    }
+    private final ColumnPolicy columnPolicy;
 
-    public DynamicReference(ReferenceIdent ident, RowGranularity granularity, int position) {
+    public DynamicReference(ReferenceIdent ident, RowGranularity granularity, int position, ColumnPolicy columnPolicy) {
         super(ident, granularity, DataTypes.UNDEFINED, position, null);
+        this.columnPolicy = columnPolicy;
```

Comment:
I'm a bit confused. Part of the motivation for moving `ColumnPolicy` to the `ObjectType` was that only objects have a policy, and regular columns don't.

If we need to add a columnPolicy to the `DynamicReference`, that's apparently not the case and we may be better off keeping it on `(Simple)Reference` ?

What's a scenario where the columnPolicy on a non-object column for dynamic matters/is different?